sentinel/tests/unit/lib/metadata.js

```diff
@@ -0,0 +1,103 @@
+const sinon = require('sinon');
+const chai = require('chai');
+const db = require('../../../src/db');
+const metadata = require('../../../src/lib/metadata');
+
+describe('metadata', () => {
+
+  afterEach(() => sinon.restore());
+
+  describe('getProcessedSeq', () => {
+
+    it('handles missing metadata doc', () => {
+      sinon.stub(db.sentinel, 'get').rejects({ status: 404 });
+      sinon.stub(db.medic, 'get').rejects({ status: 404 });
+      return metadata.getProcessedSeq(seq => {
+        chai.expect(db.sentinel.get.callCount).to.equal(1);
+        chai.expect(db.sentinel.get.args[0][0]).to.equal('_local/sentinel-meta-data');
+        chai.expect(db.medic.get.callCount).to.equal(2);
+        chai.expect(db.medic.get.args[0][0]).to.equal('_local/sentinel-meta-data');
+        chai.expect(db.medic.get.args[1][0]).to.equal('sentinel-meta-data');
+        chai.expect(seq).to.equal(0);
+      });
+    });
+
+    it('migrates old metadata doc', () => {
+      sinon.stub(db.sentinel, 'get').rejects({ status: 404 });
+      sinon.stub(db.medic, 'get')
+        .withArgs('_local/sentinel-meta-data').rejects({ status: 404 })
+        .withArgs('sentinel-meta-data').resolves({
+          _id: 'sentinel-meta-data',
+          _rev: '1',
+          processed_seq: '12'
+        });
+      sinon.stub(db.medic, 'put').resolves();
+      return metadata.getProcessedSeq(seq => {
```

Comment:
Same as above. There's also an assertion that needs to be changed, as `sentinel-meta-data` is being updated (which is correct!) not `_local/sentinel-meta-data`

Reply:
Started failing as soon as I fixed the promise call... Fixed.